### Project Title : Survey Dataset Analysis

#### Research Questions:
- Question 1: how many samples were collected on each day?
- Question 2: what proportion of the total respondents were aged less than 45?
- Question 3: create a new column in the dataframe “age_group”. This column should contain the age group the respondent belongs to. The age groups are 18-25, 25-40, 40-55 and 55+.
- Question 4: how many samples were collected for each age-group? Which age-group had the most samples?
- Question 5: what proportion of the respondents had opted for the RJD party in both the Vote_Now and the Past_Vote questions?
- Question 6: for each day of sample collection, determine the proportion of respondents who were fully satisfied with the performance of the CM. So if there were a total of 1000 samples on day 1 and 300 out of those said they were fully satisfied, then our answer for that day would be 0.3.
- Question 7: In a similar fashion create a day-wise proportion of respondents that opted fully dissatisfied with their MLA. Create a line plot of the result with date on x-axis and proportions on the y-axis.
- Question 8: create a pivot-table (or crosstab) with index as Past_Vote, Column as Vote_Now and cell values as the count of samples.
- Question 9: repeat the above question with the cell values as the sum of “weight”.
- Question 10: create a dataframe by performing a group by over age_group and calculate the count of total samples under each age_group.
- Question 11: create a dataframe by performing a group by over age_group and finding the count of total samples for each age_group that opted for the JD(U) party in Vote_Now.
- Question 12: join/merge the two dataframes from questions 10 and 11 with the common column as age_group.

In [1]:
# import libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

%matplotlib inline

#### Gathering

In [3]:
# load data into a dataframe and display the first 5 rows

df = pd.read_csv("dataset/survey_TI.csv")
df.head()

response_id collection_date urban_rural  education               income  \
0            1       5-24-2017       Rural  12th Pass  € 30,000 - € 50,000   
1            2       5-24-2017       Rural   Graduate  € 20,000 - € 30,000   
2            3       5-24-2017       Rural   Graduate   € 5,000 - € 10,000   
3            4       5-24-2017       Urban   Graduate  € 10,000 - € 20,000   
4            5       5-24-2017       Rural   Graduate  € 50,000 - € 80,000   

  Vote_Now     Past_Vote     CM_satisfaction    MLA_satisfaction age gender  \
0      BJP           BJP  Fully Dissatisfied  Fully Dissatisfied  26   Male   
1     RLSP         JD(U)  Somewhat Satisfied  Fully Dissatisfied  34   Male   
2   JAP(L)           BJP  Fully Dissatisfied  Somewhat Satisfied  25   Male   
3      RJD           RJD  Fully Dissatisfied  Somewhat Satisfied  36   Male   
4    JD(U)  Did not vote     Fully Satisfied     Fully Satisfied  22   Male   

   assembly_no category    weight  
0          211      Gen  0.346969  
1          189       BC  0.603419  
2          191      Gen  0.673762  
3          194      Gen  1.101298  
4          176       BC  0.333505

#### Question 1: how many samples were collected on each day?

In [4]:
# df.collection_date.unique()
# df.collection_date.nunique()
# df.collection_date.isnull().sum()

df.collection_date.value_counts()

5-24-2017    1479
5-25-2017     998
5-28-2017     761
5-27-2017     665
5-29-2017     620
6-1-2017      607
5-26-2017     598
5-30-2017     582
5-31-2017     557
Name: collection_date, dtype: int64

#### Question 2: what proportion of the total respondents were aged less than 45?

In [5]:
# df.age.isnull().sum()
# df.age.dtype

# df.age.unique()
df.age.nunique()

60

In [6]:
# change '24ko' to 24
df.age.replace({'24ko':'24'}, inplace = True)

In [7]:
# df.age.unique()
# df.age.value_counts()
df.age.nunique()

59

In [8]:
df.age.dtype

dtype('O')

In [9]:
# change datatype

df.age = df.age.astype(int)
df.age.dtype

dtype('int32')

In [10]:
# dataframe of participants less than 45 years
df_45 = df[df.age < 45]

In [11]:
# proportion of participants less than 45 years
prop_45 = (len(df_45)/len(df)) * 100

print(format(prop_45,".1f"))


93.2


The proportion of participants less than the age of 45 is **93.2%**

#### Question 3: create a new column in the dataframe “age_group”. This column should contain the age group the respondent belongs to. The age groups are 18-25, 25-40, 40-55 and 55+.

In [12]:
# create a copy of the original dataframe

ag_df = df.copy()
ag_df.head(2)

response_id collection_date urban_rural  education               income  \
0            1       5-24-2017       Rural  12th Pass  € 30,000 - € 50,000   
1            2       5-24-2017       Rural   Graduate  € 20,000 - € 30,000   

  Vote_Now Past_Vote     CM_satisfaction    MLA_satisfaction  age gender  \
0      BJP       BJP  Fully Dissatisfied  Fully Dissatisfied   26   Male   
1     RLSP     JD(U)  Somewhat Satisfied  Fully Dissatisfied   34   Male   

   assembly_no category    weight  
0          211      Gen  0.346969  
1          189       BC  0.603419

In [13]:
ag_df.columns

Index(['response_id', 'collection_date', 'urban_rural', 'education', 'income',
       'Vote_Now', 'Past_Vote', 'CM_satisfaction', 'MLA_satisfaction', 'age',
       'gender', 'assembly_no', 'category', 'weight'],
      dtype='object')

In [14]:
#ag_df.age.unique()

In [15]:
# create age_group column by binning the age column
# method 1

ag_df['age_group'] = pd.cut(x= ag_df['age'], 
                            bins=[18, 25, 40, 55, np.inf], 
                            labels=['18-25', '25-40', '40-55','55+'],
                            include_lowest=True, right= False)

In [16]:
# ag_df['age_group'] = ag_df['age'].apply(lambda age: '18-25' if age < 25 else '25-40' if age < 40 else '40-55' if age < 55 else '55+')


In [17]:
# method 3
# def bin_age(age):
#     if age <25:
#         return '18-25'
#     elif age <40:
#         return '25-40'
#     elif age < 55:
#         return '40-55'
#     else:
#         return '55+'
# ag_df['age_group'] = ag_df['age'].apply(bin_age)

In [18]:
move_col = ag_df.pop("age_group")

ag_df.insert(10, "age_group", move_col)

In [19]:
ag_df.head(2)

response_id collection_date urban_rural  education               income  \
0            1       5-24-2017       Rural  12th Pass  € 30,000 - € 50,000   
1            2       5-24-2017       Rural   Graduate  € 20,000 - € 30,000   

  Vote_Now Past_Vote     CM_satisfaction    MLA_satisfaction  age age_group  \
0      BJP       BJP  Fully Dissatisfied  Fully Dissatisfied   26     25-40   
1     RLSP     JD(U)  Somewhat Satisfied  Fully Dissatisfied   34     25-40   

  gender  assembly_no category    weight  
0   Male          211      Gen  0.346969  
1   Male          189       BC  0.603419

#### Question 4: how many samples were collected for each age-group? Which age-group had the most samples?

In [20]:
ag_df.age_group.value_counts()

25-40    3844
18-25    2094
40-55     801
55+       128
Name: age_group, dtype: int64

<table>
    <tr><td>Age group </td><td>Participants</td></tr>
    <tr><td>18-25  </td><td>2094</td></tr>
    <tr><td>25-40  </td><td>3844</td></tr>
    <tr><td>40-55 </td><td>801</td></tr>
    <tr><td>55+ </td><td>128</td></tr>
</table>

> Age group 25-40 had the most samples with 3844 participants.

#### Question 5: what proportion of the respondents had opted for the RJD party in both the Vote_Now and the Past_Vote questions?

In [21]:
RJD_df = df[(df.Vote_Now == 'RJD') & (df.Past_Vote == 'RJD')]

In [22]:
# proportion
prop_RJD = (len(RJD_df)/len(df)) * 100

print(format(prop_RJD,".1f"))


11.8


The proportion of the respondents had opted for the RJD party in both the Vote_Now and the Past_Vote questions is **11.8%**